plasmapy/diagnostics/langmuir.py

```diff
@@ -22,29 +23,30 @@ class Probe():
         Electrode configuration. Possible options are 'single' (default),
         'double', 'double-asym', 'triple', 'tetra', 'penta'. The electrode
         configuration must correspond to the number of areas provided.
-    
+
     """
     def __init__(self, area=u.m**2, config='single'):
         # Initialize probe properties
         self.area = area
         self.config = config
 
-def langmuir_analysis(potential_sweep, measured_current, probe, ion='H',
-                      method='maxwellian', smoothing=None, polyorder=3):
+
+def swept_probe_analysis(potential_sweep, measured_current, probe, ion='H',
```

Comment:
I think this should be particle instead of ion. Depending on how you set the potential you can measure electrons as well, and in many cases you want to just that!

Reply:
Well, the electron mass is known. What we need to know is the ion mass in order to obtain the ion density from the ion saturation current, right?

Reply:
Fair point.

It may also depend on how you view the sweep, or rather what you assume as input (this should be documented in the docstring). Is the sweep across the entire voltage from ion saturation to electron saturation, is it just the ion saturation region, or just the electron dominated region before saturation, or does the user decide by setting the particle? My initial comment was assuming this last option, but that may not be the best way to do it.

Reply:
Yes, that's something we'll have to decide. Fact is that in order to obtain some parameters (i.e. ion density, for a single probe) we need properties provided by multiple parts of the trace (ion saturation current, in the ion current dominated region, and electron temperature, which may follow from the exponential electron current region). 

Another thing to consider is that the sweep might be split up, for example due to amplification of the ion saturation current. We might need to add a function to "stitch" these together, or require the user to do that.

Reply:
Not sure how to deal with the former. The latter can be done with a pre-processing function where the user inputs the current trace and scale. They run this pre-processing function manually and then pass the current trace (which has now been standardized) into the `swept_probe_analysis` function.

Reply:
Regarding the former, I think we should provide different functions for each method / section of the analysis, so that they can be called independently. The `swept_probe_analysis` function would then call upon these functions to provide a standard "off the shelf" analysis. That way a user can either use our quick and easy analysis function, or incorporate the subfunctions in a custom analysis.